docs/addons/karpenter.md

```diff
@@ -16,66 +16,63 @@ Karpenter works by:
-      { key: 'topology.kubernetes.io/zone', op: 'NotIn', vals: ['us-west-2c']},
-      { key: 'kubernetes.io/arch', op: 'In', vals: ['amd64','arm64']},
-      { key: 'karpenter.sh/capacity-type', op: 'In', vals: ['spot','on-demand']},
-  ],
-  subnetTags: {
-    "Name": "blueprint-construct-dev/blueprint-construct-dev-vpc/PrivateSubnet1",
-  },
-  securityGroupTags: {
-    "kubernetes.io/cluster/blueprint-construct-dev": "owned",
-  },
-  taints: [{
-    key: "workload",
-    value: "test",
-    effect: "NoSchedule",
-  }],
-  amiFamily: "AL2",
-  amiSelector: {
-    "karpenter.sh/discovery/MyClusterName": '*',
-  },
-  consolidation: { enabled: true },
-  ttlSecondsUntilExpired: 2592000,
-  weight: 20,
-  interruptionHandling: true,
-  tags: {
-    schedule: 'always-on'
-  },
-  blockDeviceMappings: [{
-    deviceName: "/dev/xvda",
-    ebs: {
-      volumeSize: 100,
-      volumeType: EbsDeviceVolumeType.GP3,
-      deleteOnTermination: true
+    version: 'v0.33.1',
+    NodePoolSpec: {
+      labels: {
+          type: "karpenter-test"
+      },
+      annotations: {
+          "eks-blueprints/owner": "young"
+      },
+      taints: [{
+          key: "workload",
+          value: "test",
+          effect: "NoSchedule",
+      }],
+      requirements: [
+          { key: 'node.kubernetes.io/instance-type', operator: 'In', values: ['m5.2xlarge'] },
+          { key: 'topology.kubernetes.io/zone', operator: 'In', values: ['us-west-2a','us-west-2b', 'us-west-2c']},
+          { key: 'kubernetes.io/arch', operator: 'In', values: ['amd64','arm64']},
+          { key: 'karpenter.sh/capacity-type', operator: 'In', values: ['spot']},
+      ],
+      disruption: {
+          consolidationPolicy: "WhenEmpty",
+          consolidateAfter: "30s",
+          expireAfter: "20m",
+          budgets: [{nodes: "10%"}]
+      }
+    },
+    EC2NodeClassSpec: {
+      amiFamily: "AL2",
+      subnetSelectorTerms: [{ tags: { "Name": "my-stack-name/my-stack-name-vpc/PrivateSubnet*" }}],
+      securityGroupSelectorTerms: [{ tags: { "aws:eks:cluster-name": "my-stack-name" }}],
     },
-  }],
+    interruptionHandling: true,
 });
 
 const blueprint = blueprints.EksBlueprint.builder()
-  .version("auto")
+  .version(KubernetesVersion.V1_28)
   .addOns(karpenterAddOn)
   .build(app, 'my-stack-name');
 ```
 
-The add-on automatically sets the following Helm Chart [values](https://github.com/aws/karpenter/tree/main/charts/karpenter#values), and it is **highly recommended** not to pass these values in (as it will result in errors):
-- settings.aws.defaultInstanceProfile
-- settings.aws.clusterEndpoint
-- settings.aws.clusterName
-- settings.aws.interruptionQueueName (if interruption handling is enabled)
+The add-on automatically sets the following Helm Chart [values](https://github.com/aws/karpenter-provider-aws/tree/main/charts/karpenter#values), and it is **highly recommended** not to pass these values in (as it will result in errors):
```

Comment:
why would passing the values result in errors? if the values are correct, it should succeed.

Reply:
I can change the wording to recommend not to pass them in

Reply:
done